src/bedriftsmeny/byggOrganisasjonsTre.ts

```diff
@@ -0,0 +1,94 @@
+): Promise<JuridiskEnhetMedUnderEnheterArray[]> {
+    const juridiskeEnheter = organisasjoner.filter(function(organisasjon: Organisasjon) {
+        return organisasjon.Type === 'Enterprise';
+    });
+    const underenheter = organisasjoner.filter(function(organisasjon: Organisasjon) {
+        return organisasjon.OrganizationForm === 'BEDR' && organisasjon.OrganizationNumber;
+    });
+    const jurEnheterOrgNr = juridiskeEnheter.map(jurorg => jurorg.OrganizationNumber);
+    const underEnheterMedJuridiskEnhet = organisasjoner.filter(org => {
+        return jurEnheterOrgNr.includes(org.ParentOrganizationNumber);
+    });
+
+    const underEnheterUtenJuridiskEnhet = organisasjoner.filter(org => {
+        return !underEnheterMedJuridiskEnhet.includes(org) && org.OrganizationForm === 'BEDR';
+    });
+    const finnJuridiskeEnheter = async (underEnheterUtenJuridisk: Organisasjon[]) => {
+        const juridiskeEnheterUtenTilgang: Organisasjon[] = await hentAlleJuridiskeEnheter(
+            underEnheterUtenJuridisk.map(org => org.ParentOrganizationNumber)
+        );
+        return juridiskeEnheterUtenTilgang;
+    };
+    if (underEnheterUtenJuridiskEnhet.length>0) {
+        await finnJuridiskeEnheter(underEnheterUtenJuridiskEnhet).then(juridiskeEnheterUtenTilgang => {juridiskeEnheter.push(...juridiskeEnheterUtenTilgang);
+        });
+    }
+    const orgtre = settSammenJuridiskEnhetMedUnderOrganisasjoner(juridiskeEnheter, underenheter);
+    return orgtre.sort((a, b) => a.JuridiskEnhet.Name.localeCompare(b.JuridiskEnhet.Name));
+}
+
+const settSammenJuridiskEnhetMedUnderOrganisasjoner = (
+    juridiskeEnheter: Organisasjon[],
+    underEnheter: Organisasjon[]
+): JuridiskEnhetMedUnderEnheterArray[] => {
+    const organisasjonsTre: JuridiskEnhetMedUnderEnheterArray[] = juridiskeEnheter.map(juridiskEnhet => {
+        const underenheter = underEnheter.filter(
+            underenhet => underenhet.ParentOrganizationNumber === juridiskEnhet.OrganizationNumber
+        );
+        const resultat = {
+            JuridiskEnhet: juridiskEnhet,
+            Underenheter: underenheter
+        };
+        return resultat;
+    });
+    return organisasjonsTre.filter(orgtre => orgtre.Underenheter.length > 0);
+};
+
+export async function hentAlleJuridiskeEnheter(listeMedJuridiskeOrgNr: string[]): Promise<Organisasjon[]> {
+    const listerMedDefinerteOrgNr = listeMedJuridiskeOrgNr.filter(orgnr => {
+        return orgnr !== null;
+    });
+    let url: string = 'https://data.brreg.no/enhetsregisteret/api/enheter/?organisasjonsnummer=';
+    const distinkteJuridiskeEnhetsnr: string[] = listerMedDefinerteOrgNr.filter(
+        (jurOrg, index) => listeMedJuridiskeOrgNr.indexOf(jurOrg) === index
+    );
+    distinkteJuridiskeEnhetsnr.forEach(orgnr => {
+        if (distinkteJuridiskeEnhetsnr.indexOf(orgnr) === 0) {
+            url += orgnr;
+        } else {
+            url += ',' + orgnr;
+        }
+    });
+    let respons = await fetch(url);
```

Comment:
Her gjøres fetch-kall inni bedriftsmenyen. Jeg synes dette er en god idé, og kanskje burde vi gjøre det samme med Altinn-organisasjonene også. Men jeg har tatt i bruk komponenter med innebygde fetch-kall før, og har irritert meg over at jeg ikke f.eks. kan endre URL-en som blir kalt. F.eks. hvis jeg har min egen mock-enhetsregister jeg vil kalle lokalt, eller noe annet i testmiljø. Har du gjort noen tanker om det? 🤔 Vi kunne f.eks. ha gitt muligheten til å konfigurere opp url til enhetsregisteret selv.

Reply:
ja, jeg tenkte det kunne være et problem, og syns det er god løsning. Hvis man har et eget mock-register man vil kalle, kan man sende inn en url, så kan den være optional og at vi bruker den vi har som default

Reply:
har endret mening. Da bør man lage en komplett liste med juridiske enheter og underenheter, for det er en bedre løsning som reduserer risikoen for at appen kræsjer pga feil url-input